### PR TITLE
interface: update traffic manager APIs to support splitting weight behind serviceImport

### PR DESCRIPTION
**Pull request by zhiying-lin**
**What type of PR is this?**

/kind feature
/kind api-change

**What this PR does / why we need it**:

- add resource id in the status of endpoint and profile
- update weight to follow gateway api. Default to 1 and allow for 0.
- display the original weight from the cluster in the endpoint
- add weight on the serviceExport

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:

- [ ] run `make reviewable` for basic local test
- [ ] Read and followed fleet-networking's [Code of conduct](https://github.com/Azure/fleet-networking/blob/main/CODE_OF_CONDUCT.md).
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

### CODE REVIEW

File: api/v1alpha1/serviceexport_types.go [hunk 1]
```diff
@@ -9,6 +9,27 @@ import (
...
+// ServiceExportSpec describes an exported service extra information.
+type ServiceExportSpec struct {
+	// exportedLabels describes the labels exported.
+	// +optional
+	ExportedLabels map[string]string `json:"exportedLabels,omitempty"`
+	// exportedAnnotations describes the annotations exported.
+	// Possible Annotations: "weight".
+	// "weight" specifies the proportion of requests forwarded to the cluster within a serviceImport.
+	// This is computed as weight/(sum of all weights in the serviceImport).
+	// If weight is set to 0, no traffic should be forwarded for this entry. If unspecified, weight defaults to 1.
```
Comment (ryanzhang-oss) [inline api/v1alpha1/serviceexport_types.go:26]: so we decide to not support the less than average traffic to a cluster while split between the rest case?

File: api/v1alpha1/trafficmanagerbackend_types.go [hunk 2]
```diff
@@ -75,18 +82,33 @@ type TrafficManagerEndpointStatus struct {
 	// +required
 	Name string `json:"name"`
 
+	// ResourceID is the fully qualified Azure resource Id for the resource.
+	// Ex - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/trafficManagerProfiles/{profileName}/azureEndpoints/{name}
+	ResourceID string `json:"resourceID,omitempty"`
```
Comment (ryanzhang-oss) [inline api/v1alpha1/trafficmanagerbackend_types.go:87]: I think this is the same as the one in the TrafficManagerProfile?  or this is the one for the endpoint (IP)?

File: api/v1alpha1/trafficmanagerbackend_types.go [hunk 3]
```diff
@@ -75,18 +82,33 @@ type TrafficManagerEndpointStatus struct {
 	// +required
 	Name string `json:"name"`
 
+	// ResourceID is the fully qualified Azure resource Id for the resource.
+	// Ex - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/trafficManagerProfiles/{profileName}/azureEndpoints/{name}
```
Comment (ryanzhang-oss) [inline api/v1alpha1/trafficmanagerbackend_types.go:86]: this comment seems to indicate that the resourceID is the ATMs?

File: api/v1alpha1/trafficmanagerbackend_types.go [hunk 4]
```diff
@@ -75,18 +82,33 @@ type TrafficManagerEndpointStatus struct {
...
+	// Possible values are from 0 to 1000.
 	// +optional
 	Weight *int64 `json:"weight,omitempty"`
 
 	// The fully-qualified DNS name or IP address of the endpoint.
 	// +optional
 	Target *string `json:"target,omitempty"`
 
-	// Cluster is where the endpoint is exported from.
+	// From is where the endpoint is exported from.
```
Comment (ryanzhang-oss) [inline api/v1alpha1/trafficmanagerbackend_types.go:98]: why change this field name? Is there some API pattern we follow ?

File: api/v1alpha1/trafficmanagerbackend_types.go (hunk 3 above)
Comment (zhiying-lin) [reply api/v1alpha1/trafficmanagerbackend_types.go:86]: yes, we'll show the Azure resource ID based on the wallow feedback for both profile and endpoint.

we'll continue to use uuid in ATM resource names to solve name conflicts issue.

File: api/v1alpha1/trafficmanagerbackend_types.go (hunk 2 above)
Comment (zhiying-lin) [reply api/v1alpha1/trafficmanagerbackend_types.go:87]: this one is for azure traffic manager endpoint.

File: api/v1alpha1/trafficmanagerbackend_types.go (hunk 4 above)
Comment (zhiying-lin) [reply api/v1alpha1/trafficmanagerbackend_types.go:98]: i'm trying to solve duplicate field "cluster" before v1beta1, meanwhile we're introducing two layers weight and i created a structure to include the "weight" per cluster for better visibility.

image

**Review by ryanzhang-oss — APPROVED** [review]

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**